lintlizard/__init__.py

```diff
@@ -34,38 +35,72 @@ def version_command(self) -> Command:
 
 TOOLS = [
     CommandTool('flake8'),
-    CommandTool('isort', run_params=('-c', '.'), fix_params=('.',)),
-    CommandTool('mypy'),
-    CommandTool('black', run_params=('--check', '.'), fix_params=('.',)),
+    CommandTool('isort', run_params=('-c',), fix_params=tuple()),
+    CommandTool('mypy', allow_specify_files=False),
+    CommandTool('black', run_params=('--check',), fix_params=tuple()),
 ]
 
 
-def execute_tools(fix: bool) -> Iterable[bool]:
+def execute_tools(fix: bool, files: Tuple[str, ...]) -> Iterable[bool]:
     tools = TOOLS
     if fix:
         tools = [tool for tool in tools if tool.fixable]
     for tool in tools:
         print('*' * 79)
         try:
             subprocess.run(args=tool.version_command(), check=True)
-            cmd = tool.fix_command() if fix else tool.run_command()
-            subprocess.run(args=cmd, check=True)
+            cmd = (
+                tool.fix_command()
+                if fix and tool.fix_params is not None
+                else tool.run_command()
+            )
+            subprocess.run(args=cmd + files, check=True)
         except subprocess.CalledProcessError:
             yield False
         else:
             yield True
 
 
+def get_changed_files() -> Iterable[str]:
+    try:
+        subprocess.run(args=['git', '--version'], check=True)
+        result = subprocess.run(
+            args=[
+                "git",
+                "diff",
+                "--name-only",
+                "--cached",
+                "--diff-filter=d",
+                "HEAD",
+                "--",
+                "*.py",
+            ],
+            stdout=subprocess.PIPE,
+        )
+        return result.stdout.decode('utf8').strip().split(' ')
+    except subprocess.CalledProcessError:
+        raise Exception('Error encountered when determining changed files.')
+
+
 def main() -> None:
     args = make_arg_parser().parse_args()
-    tool_results = list(execute_tools(fix=args.fix))
+    files = args.files or []
+    if args.changed:
+        files.extend(get_changed_files())
+
+    if not files:
```

Comment:
I would prefer that if you asked for `--changed` but nothing's actually changed, it should skip the checks? You could put it into a commit hook with `--changed` and avoid the extra wait if all you did is some `readme.md` tweak.